### PR TITLE
feat(sms): support for templates for the sms message

### LINKED ISSUE #196: Implement SMS Templates for Auth
Currently the SMS that goes out says "Your code is 000000"

Ideally we are able to customize a template and insert the code as a variable:

*Nhost - Here is your code: {{verification_code}}. Don't share it with anyone*

To push things even further, you could even make Auth variables available for the templates:

*{{first_name}}, here is your code: {{verification_code}}. Don't share it with anyone*

### PR DESCRIPTION
**Pull request by L3K0V**
Before submitting this PR:

### Checklist

- [x] No breaking changes
- [x] Tests pass
- [x] New features have new tests
- [x] Documentation is updated

### Breaking changes

Avoid breaking changes and regressions. If you feel it is unavoidable, make it explicit in your PR comment so we can review it and see how to handle it.

### Tests

- please make sure your changes pass the current tests (Use the `make test` or the `make watch` command).
- if you are introducing a new feature, please write as much tests as possible.

### Documentation

Please make sure the documentation is updated accordingly, in particular:

- [Workflows](https://github.com/nhost/hasura-auth/tree/main/docs/workflows). Workflows are [Mermaid sequence diagrams](https://mermaid-js.github.io/mermaid/#/sequenceDiagram)
- [Schema](https://github.com/nhost/hasura-auth/blob/main/docs/schema.md). The schema in a [Mermaid ER diagram](https://mermaid-js.github.io/mermaid/#/entityRelationshipDiagram)
- [Environment variables](https://github.com/nhost/hasura-auth/blob/main/docs/environment-variables.md). Please adjust the [.env.example](https://github.com/nhost/hasura-auth/blob/main/.env.example) file accordingly
- OpenApi specifications. We are using inline [JSDoc annotations](https://www.npmjs.com/package/express-jsdoc-swagger)

### Conventional commits

Versioning and changelog are generated following [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/). Please do your best to follow the convention, but don't worry about it too much. We will be most likely moving away to [changesets](https://github.com/changesets/changesets) soon.

Closes #196

### CODE REVIEW

**Comment by L3K0V** [comment]
@elitan can you check this?

**Comment by L3K0V** [comment]
@elitan @plmercereau any ideas when this could be reviewed, merged and shipped?

**Comment by smiziara** [comment]
> @elitan @plmercereau any ideas when this could be reviewed, merged and shipped?

We are needing this PR merged too, we cannot send SMS in english.

**Comment by elleryfamilia** [comment]
@elitan Can we prioritize merging this?

**Comment by elitan** [comment]
Sorry for the delay, I will review this PR next week. I've put time in my calendar for it.

File: test/routes/misc/templates.test.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+  it('should render proper sms template', async () => {
+    const template = 'signin-passwordless-sms';
+    const message = await renderTemplate(`${template}/text`, {
+      locale: 'en',
+      displayName: 'John Doe',
+      code: '123456',
+    });
+
+    expect(message).toEqual('Your code is 123456.');
+  });
```
Comment (elitan) [inline test/routes/misc/templates.test.ts:13]: Please add a test for `locale: 'fr'` too to ensure we're not always using the fall-back message if `message` is empty.

**Review by elitan — CHANGES_REQUESTED** [review]
A small request, then this one is ready to be merged!

Thank you.

File: test/routes/misc/templates.test.ts (hunk 1 above)
Comment (L3K0V) [reply test/routes/misc/templates.test.ts:13]: Done! Looking forward! Thanks!

**Review by elitan — APPROVED** [review]
